src/Polly.Core.Tests/Retry/RetryResilienceStrategyBuilderExtensionsTests.cs

```diff
@@ -4,25 +4,41 @@
 
 namespace Polly.Core.Tests.Retry;
 
+#pragma warning disable CA2012 // Use ValueTasks correctly
+
 public class RetryResilienceStrategyBuilderExtensionsTests
 {
     public static readonly TheoryData<Action<ResilienceStrategyBuilder>> OverloadsData = new()
     {
         builder =>
         {
-            builder.AddRetry(retry=>retry.Result(10));
+            builder.AddRetry(retry=>retry.HandleResult(10));
             AssertStrategy(builder, RetryBackoffType.Exponential, 3, TimeSpan.FromSeconds(2));
         },
         builder =>
         {
-            builder.AddRetry(retry=>retry.Result(10), RetryBackoffType.Linear);
+            builder.AddRetry(retry=>retry.HandleResult(10), RetryBackoffType.Linear);
             AssertStrategy(builder, RetryBackoffType.Linear, 3, TimeSpan.FromSeconds(2));
         },
         builder =>
         {
-            builder.AddRetry(retry=>retry.Result(10), RetryBackoffType.Linear, 2, TimeSpan.FromSeconds(1));
+            builder.AddRetry(retry=>retry.HandleResult(10), RetryBackoffType.Linear, 2, TimeSpan.FromSeconds(1));
             AssertStrategy(builder, RetryBackoffType.Linear, 2, TimeSpan.FromSeconds(1));
         },
+        builder =>
+        {
+            builder.AddRetry(retry => retry.HandleResult(10), attempt =>
+            {
+                return TimeSpan.FromMilliseconds(attempt);
+            });
```

Comment:
Maybe use the terser syntax?
```suggestion
            builder.AddRetry(retry => retry.HandleResult(10), attempt => TimeSpan.FromMilliseconds(attempt));
```

Reply:
Done.